specutils/analysis/template_comparison.py

```diff
@@ -0,0 +1,138 @@
+    elif resample_method == "spline_interpolated":
+        return SplineInterpolatedResampler()
+    else:
+        return None
+
+def _template_match(observed_spectrum, template_spectrum, resample_method):
+    """
+    Resample the template spectrum to match the wavelength of the observed spectrum.
+    Then, calculate chi2 on the flux of the two spectra.
+
+    Parameters
+    ----------
+    observed_spectrum : :class:`~specutils.Spectrum1D`
+        the observed spectrum
+    template_spectrum : :class:`~specutils.Spectrum1D`
+        the template spectrum, which will be resampled to match the wavelength of the observed_spectrum
+
+    Returns
+    -------
+    normalized_template_spectrum : :class:`~specutils.Spectrum1D`
+        the template_spectrum that has been normalized
+    chi2 : `float`
+        the chi2 of the flux of the observed_spectrum and the flux of the normalized_template_spectrum
+    """
+    # Resample template
+    if _resample(resample_method) != 0:
+        fluxc_resample = _resample(resample_method)
+        template_obswavelength = fluxc_resample(template_spectrum, observed_spectrum.wavelength)
+
+    # Normalize spectra
+    normalization = _normalize_for_template_matching(observed_spectrum, template_obswavelength)
+
+    # Numerator
+    num_right = normalization*template_obswavelength.flux
+    num = observed_spectrum.flux - num_right
+
+    # Denominator
+    denom = observed_spectrum.uncertainty.array * observed_spectrum.flux.unit
+
+    # Get chi square
+    result = (num/denom)**2
+    chi2 = np.sum(result)
+
+    # Create normalized template spectrum, which will be returned with corresponding chi2
+    normalized_template_spectrum = Spectrum1D(spectral_axis=template_spectrum.spectral_axis,
+                                              flux=template_spectrum.flux*normalization)
+
+    return normalized_template_spectrum, chi2
+
+def template_match(observed_spectrum, spectral_templates, resample_method="flux_conserving"):
+    """
+    Find which spectral templates is the best fit to an observed spectrum by computing the chi-squared. If two
+    template_spectra have the same chi2, the first template is returned
+
+    Parameters
+    ----------
+    observed_spectrum : :class:`~specutils.Spectrum1D`
+        the observed spectrum
+    spectral_templates : :class:`~specutils.Spectrum1D` or :class:`~specutils.SpectrumCollection` or `list` or anything
+        that will give a single :class:`~specutils.Spectrum1D` when iterated over.
+        The template spectra, which will be resampled and normalized and compared to the observed_spectrum, where the
+        smallest chi2 and normalized_template_spectrum will be returned
+
+    Returns
+    -------
+    normalized_template_spectrum : :class:`~specutils.Spectrum1D`
+        the template_spectrum that has been normalized
+    chi2 : `float`
+        the chi2 of the flux of the observed_spectrum and the flux of the normalized_template_spectrum
```

Comment:
This needs to also now have `smallest_chi_index` listed here as the third return